src/axom/core/Array.hpp

```diff
@@ -966,8 +970,8 @@ class Array : public ArrayBase<T, DIM, Array<T, DIM, SPACE>>
   IndexType m_num_elements = 0;
   IndexType m_capacity = 0;
   double m_resize_ratio = DEFAULT_RESIZE_RATIO;
-  int m_allocator_id;
-  bool m_executeOnGPU;
+  int m_allocator_id = INVALID_ALLOCATOR_ID;
+  bool m_executeOnGPU {false};
```

Comment:
Why do we use different variable initialization syntax? Do we have a convention that we follow?

Reply:
I'm not aware of a convention. Point taken though. This file is using assignment-style initialization for the other members so I'll change to that for the bool.

Reply:
@publixsubfan -- I think in the past, you had a reason to prefer the curly brace initialization syntax. Do you recall what it was? 

Once this is resolved, it might be good to update our coding conventions w/ a preference. At the very least, we should be consistent w/in a single file.